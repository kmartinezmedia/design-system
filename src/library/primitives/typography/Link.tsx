import React, { FunctionComponent } from 'react';
import { Text, TextProps } from './Text';

const Link: FunctionComponent<TextProps> = ({
  children,
  color = 'primary',
  ...textProps
}) => (
  <Text color={color} {...textProps}>
    {children}
  </Text>
);

export { Link };
