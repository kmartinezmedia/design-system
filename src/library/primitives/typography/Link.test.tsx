import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Link } from './Link';

describe('Link', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Link>Hey, lil 🐛</Link>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
