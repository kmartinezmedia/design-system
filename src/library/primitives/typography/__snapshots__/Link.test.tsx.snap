// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link default 1`] = `
<Text
  style={
    Array [
      Object {},
      Object {
        "color": "#1652F0",
        "lineHeight": undefined,
        "textAlign": undefined,
      },
      Object {},
      Object {},
      Object {},
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
